controllers/update_pods.go

```diff
@@ -164,6 +173,22 @@ func shouldRequeueDueToTerminatingPod(pod *corev1.Pod, cluster *fdbv1beta2.Found
 		!cluster.ProcessGroupIsBeingRemoved(processGroupID)
 }
 
+// Because the internal.fault_tolerance relies in fdbcli status only we need to make sure it is
+// synced with the cluster status to prevent race conditions.
+func shouldRequeueDueToClusterStatusDrift(cluster *fdbv1beta2.FoundationDBCluster, adminClient fdbadminclient.AdminClient, podMap map[string]*corev1.Pod) (bool, error) {
+	status, err := adminClient.GetStatus()
+	if err != nil {
+		return false, err
+	}
+	//Check that all processes in fdbcli status have a matching pod.
+	for _, p := range status.Cluster.Processes {
+		if _, ok := podMap[p.Locality["instance_id"]]; !ok {
```

Comment:
```suggestion
		if _, ok := podMap[p.Locality[fdbv1beta2.FDBLocalityInstanceIDKey]; !ok {
```

See: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/api/v1beta2/foundationdb_labels.go#L62